ring/lifecycler.go

```diff
@@ -63,12 +68,17 @@ func (cfg *LifecyclerConfig) RegisterFlagsWithPrefix(prefix string, f *flag.Flag
 		prefix = "ingester."
 	}
 
+	// Allow clients to override default value by setting it before calling this method.
+	if cfg.FinalSleep == 0 {
```

Comment:
Hmm, 0 value for this check may actually not work that well... user (or in our case our package) may want to set it to 0 to *disable* final sleep.

Reply:
I think we might need to bite the bullet, and set the default to 0 in dskit, but let it respect non-zero value when registering flag.

Reply:
technically the client can set it to a very short duration (1 ns). That will be the minimal sleep timer supported by the OS/CPU ([usually 1-2 _ms_](https://github.com/golang/go/issues/29485)). It's essentially disabled then.

Reply:
actully a better proposition. `time.Duration` is `int64`, so we can also accept `-1` as the disabled value. go's `time.Sleep` docs say that negative and zero values cause `time.Sleep` [to return immediately](https://pkg.go.dev/time#Sleep).

Reply:
> technically the client can set it to a very short duration (1 ns). That will be the minimal sleep timer supported by the OS/CPU ([usually 1-2 _ms_](https://github.com/golang/go/issues/29485)). It's essentially disabled then.

I don't think asking user to set it to 1ms to disable is a good user experience.

> time.Duration is int64, so we can also accept -1 as the disabled value. go's time.Sleep docs say that negative and zero values cause time.Sleep to return immediately.

I considered this option, but the problem is that if user passes "0", he doesn't get "no wait", but 30 seconds instead. It's this weird behaviour that I'd like to avoid.

Reply:
so setting a negative value for the struct field will disable sleep (will still want to do an `if FinalSleep < 0 { FinalSleep = 0 }`) and `0` will be treated as unset and `=30s`

Reply:
I propose that we actually change the default in `dskit` but also do:

```
f.DurationVar(&cfg.FinalSleep, prefix+"final-sleep", cfg.FinalSleep, "Duration to sleep for before exiting, to ensure metrics are scraped.")
```

This will also respect other values, set by code using `dskit`, so others can go back to 30s if they want. Negative and 0 values will cause no sleep.

Reply:
I agree that the `0 = 30s` is bad UX. Pushed in 
https://github.com/grafana/dskit/pull/121/commits/57a7fb141493ad06c0dac17f1ab0d31d823eecbb
the changes to remove that `if`